cursive/examples/filters.rs

```diff
@@ -0,0 +1,135 @@
+use cursive::view::Resizable;
+use cursive::views::{Dialog, LinearLayout, RadioGroup, TextView};
+
+#[derive(Clone, Copy, Debug)]
+enum Color {
+    Red,
+    Blue,
+}
+
+#[derive(Clone, Copy, Debug)]
+enum Size {
+    Large,
+    Small,
+}
+
+#[derive(Default)]
+struct Filters {
+    color: Option<Color>,
+    size: Option<Size>,
+}
+
+// This is an examples of a Settings/Configuration "pop-up" where we can
+// modify user_data to be used elsewhere in our application.
+
+fn main() {
+    let mut siv = cursive::default();
+    siv.set_user_data(Filters::default());
+    // Leave the menu visible
+    siv.set_autohide_menu(false);
+    siv.menubar()
+        .add_leaf("Filters", |s| {
+            // Hide menu while "pop-up" settings are visible
+            s.set_autohide_menu(true);
+
+            let mut color_group: RadioGroup<Option<Color>> = RadioGroup::new();
+            let mut size_group: RadioGroup<Option<Size>> = RadioGroup::new();
+
+            // Get current filters to draw the correct selected buttons.
+            let user_data = s.user_data::<Filters>().unwrap();
+            let current_color = user_data.color;
+            let current_size = user_data.size;
+            s.add_layer(
+                Dialog::new()
+                    .title("Filters")
+                    .content(
+                        LinearLayout::vertical()
+                            .child(
+                                LinearLayout::horizontal()
+                                    .child(
+                                        TextView::new("Color:")
+                                            .fixed_width(15)
+                                            .fixed_height(2),
+                                    )
+                                    .child(
+                                        color_group
+                                            .button(None, "Any")
+                                            .fixed_width(10),
+                                    )
+                                    .child({
+                                        let mut button = color_group
+                                            .button(Some(Color::Red), "Red");
+                                        if let Some(Color::Red) = current_color
+                                        {
+                                            button.select();
+                                        }
+                                        button.fixed_width(10)
+                                    })
```

Comment:
Nit: you can use [`With`](https://docs.rs/cursive/latest/cursive/trait.With.html) for that:
```suggestion
                                    .child(
                                        color_group
                                            .button(Some(Color::Red), "Red")
                                            .with_if(current_color == Some(Color::Red), RadioButton::select)
                                            .fixed_width(10))
                                    )
```

Reply:
`RadioButton::select` returns `EventResult`; so I had to change it for `|button| {button.select(); }`